parquet-column/src/main/java/org/apache/parquet/column/values/factory/ValuesWriterFactoryParams.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.parquet.column.values.factory;
+
+import org.apache.parquet.bytes.ByteBufferAllocator;
+import org.apache.parquet.column.ParquetProperties.WriterVersion;
+
+/**
+ * Encapsulates parameters needed to create new ValuesWriter classes as part of
+ * ValuesWriterFactory
+ */
+public class ValuesWriterFactoryParams {
+
+  public ValuesWriterFactoryParams(
+    WriterVersion writerVersion,
```

Comment:
Yeah so these values will be needed by any factory that is creating ValuesWriters as these values are needed to set up a ValuesWriter. A lot of these are pretty much constants we're reading from config (like enableDictionary / writerVersion) and wrapping up to encapsulate in one object. 

Agree, that it is possible that in the future we might have a ValueWriterFactory that wants to override some of these values (like the slabSize). Don't think this approach precludes that though. You'll end up getting the default params and you can read whatever you want from Config and then the factory can choose to use them however..